app/scripts/controllers/permissions/methodMiddleware.js

```diff
@@ -91,6 +91,12 @@ module.exports = function createRequestMiddleware ({
         }
         return
 
+      // returns permitted and installed plugins to the caller
+      case 'wallet_getPlugins':
+
+        res.result = getPlugins(req.origin)
```

Comment:
Oh, I see you're passing in `origin` here. I guess this is basically equivalent. I wouldn't block over that other note, although it seems like it's generally safer to attenuate a function before passing it somewhere else.

Reply:
Yeah, the origin middleware at the top of the stack should take care of it, but there is as you point out no reason to not attenuate the function. Did the same for `installPlugins`!